consensus/replay_test.go

```diff
@@ -278,42 +281,194 @@ var (
 // 2 - save block and committed but state is behind
 var modes = []uint{0, 1, 2}
 
+// This actually is not a test, it's for storing validator change tx data for testHandshakeReplay
```

Comment:
Does it need to look like a test then? Why can't we have a function that returns a struct with all the necessary fields instead of further clutering the global namespace?

```
type testSim struct {
  GenesisState sm.State
  Config *cfg.Config
  Chain []*types.Block
  Commits []*types.Commit
  CleanupFunc cleanupFunc
}
```

Reply:
Yeah, I think I could do it. But the orignial test containing some WAL generating chain process makeBlockchainFromWAL, isn't that also unreasonable? It needs to be executed every time call the testHandshakeReplay, totally 4*3=12 times.